### PR TITLE
> 2048 chars worked with gaps

### LINKED ISSUE #13: Create a pagination system for strings larger then 2048 characters.
**Is your feature request related to a problem? Please describe.**
Discord has a limit of 2048 characters in a text field. There is ways to extend it a little bit more then that for each embed. However eventually strings will be exceeded and pagination is needed.

**Describe the solution you'd like**
Have a speech button 💬 to press to toggle between pages of text. This is needed in the video description (player) and in the lyrics modules.

**Describe alternatives you've considered**
Add more of the text field in other sections of the embed. (around 1000 more characters can be added)

https://github.com/borisliao/mbox/blob/674f1eaf3cc4a0d0607239aa69bfb49eb1a930cf/src/music/player.py#L209
https://github.com/borisliao/mbox/blob/674f1eaf3cc4a0d0607239aa69bfb49eb1a930cf/src/music/element/Lyrics.py#L58

### PR DESCRIPTION
**Pull request by bakhoon**
I kind of split the line and then added the lyrics by gaps between.

So, I was able to get the first 2048 chars on the description and the next 1024 chars to the new text fields.

I also tested with 512 chars for both description and text fields to see if multiple text fields are added. (It actually worked).

The two worries I am having are
1. The algorithm I used because there could be better way to think optimally. 
2. What if there is a song that has a lyric without a gap. (was unable to test)
     For this part, I temporary removed the gap on the list and it worked out fine. It has a gap on the lyric display which needs to be fixed later on.

### CODE REVIEW

**Comment by Karan-Jain** [comment]
Need to discuss in a meeting. Maybe you could give us a demo.

**Comment by borisliao** [comment]
Results look good. [Image](https://user-images.githubusercontent.com/16157348/121962815-3d48ab00-cd37-11eb-9999-86d95b9a5346.png)

**Comment by Exsects** [comment]
@bakhoon Set up some time with the team to go over your changes and see where we can make improvements.

**Comment by AvJain** [comment]
Changes look ok but lets discuss in meeting, I would for example split the logic for lyrics with gap and lyrics without gap into smaller methods of their own

I also notice that you are reusing the len function multiple times, it would instead be easier to maybe just create a variable to hold that information and then just reference the variable in your code.

**Comment by bakhoon** [comment]
When you play a song with more than 2048 and then back to a song with less than 2048, it causes an error. Need to fix this part.

File: src/commander/element/LyricsEmbed.py [hunk 1]
```diff
@@ -20,11 +19,74 @@ def __init__(self, **kwargs) -> None:
...
+    
+    def get_lyric(self):
+        self.description_lyric = ''
+        self.embed_field_lyric = ''
+
+        lyric_list_by_gap = self.get_split_lyric_by_gap()
+        lyric_list_by_line = self.get_split_lyric_by_line()
+        split_type = self.get_split_type(lyric_list_by_gap)
+        
+        if(self.has_gap(lyric_list_by_gap) == True):
```
Comment (Karan-Jain) [inline src/commander/element/LyricsEmbed.py]: Remove parenthesis

File: src/commander/element/LyricsEmbed.py [hunk 2]
```diff
@@ -20,11 +19,74 @@ def __init__(self, **kwargs) -> None:
...
+            self.split_lyric(lyric_list_by_gap, split_type)
+        else:
+            self.split_lyric(lyric_list_by_line, split_type)
+            
+
+    def split_lyric(self, lyric_list, split_type):
+        for lyric_line in lyric_list:
+            if(self.embed_field_lyric == ''):
```
Comment (Karan-Jain) [inline src/commander/element/LyricsEmbed.py]: Remove parenthesis

File: src/commander/element/LyricsEmbed.py [hunk 3]
```diff
@@ -20,11 +19,74 @@ def __init__(self, **kwargs) -> None:
...
+        if(len(self.embed_field_lyric) + len(lyric_line) < self.max_textfield ):
+            self.embed_field_lyric = self.embed_field_lyric + split_type + lyric_line
+                
+        else:
+            self.add_field(name='\u200B', value=self.embed_field_lyric, inline=False)
+            self.embed_field_lyric = lyric_line
+
+
+    def get_split_type(self, lyric_list: list) -> str:
+        if(len(lyric_list) > 1):
```
Comment (Karan-Jain) [inline src/commander/element/LyricsEmbed.py]: Remove Parenthesis for the if statement directly above and the rest below

**Review by Karan-Jain — APPROVED** [review]

**Review by AvJain — APPROVED** [review]

**Pull request merged**